app/com/arpnetworking/metrics/portal/alerts/impl/DailyPartitionCreator.java

```diff
@@ -0,0 +1,283 @@
+    private final Schedule _schedule;
+    private Optional<Instant> _lastRun;
+
+    private DailyPartitionCreator(
+            final EbeanServer ebeanServer,
+            final PeriodicMetrics periodicMetrics,
+            final String schema,
+            final String table,
+            final Duration scheduleOffset,
+            final int lookahead
+    ) {
+        this(ebeanServer, periodicMetrics, schema, table, scheduleOffset, lookahead, Clock.systemUTC());
+    }
+
+    /* package private */ DailyPartitionCreator(
+            final EbeanServer ebeanServer,
+            final PeriodicMetrics periodicMetrics,
+            final String schema,
+            final String table,
+            final Duration scheduleOffset,
+            final int lookahead,
+            final Clock clock
+    ) {
+        _ebeanServer = ebeanServer;
+        _periodicMetrics = periodicMetrics;
+        _lookahead = lookahead;
+        _schedule = new PeriodicSchedule.Builder()
+                .setOffset(scheduleOffset)
+                .setPeriod(ChronoUnit.DAYS)
+                .setRunAtAndAfter(Instant.MIN)
+                .setZone(ZoneOffset.UTC)
+                .build();
+        _lastRun = Optional.empty();
+        _schema = schema;
+        _table = table;
+        _clock = clock;
+    }
+
+    /**
+     * Create {@link Props} for this actor.
+     *
+     * @param ebeanServer the ebean server
+     * @param periodicMetrics metrics instance to use
+     * @param schema The database schema name
+     * @param table The parent table name
+     * @param scheduleOffset Execution offset from midnight
+     * @param lookahead maximum number of partitions to create in advance
+     * @return A new Props.
+     */
+    public static Props props(
+            final EbeanServer ebeanServer,
+            final PeriodicMetrics periodicMetrics,
+            final String schema,
+            final String table,
+            final Duration scheduleOffset,
+            final int lookahead
+    ) {
+        return Props.create(
+                DailyPartitionCreator.class,
+                () -> new DailyPartitionCreator(
+                    ebeanServer,
+                    periodicMetrics,
+                    schema,
+                    table,
+                    scheduleOffset,
+                    lookahead
+                )
+        );
+    }
+
+    /**
+     * Ask the actor referenced by {@code ref} to start execution.
+     * <p>
+     * This will execute partition creation exactly once and then periodically thereafter using the actor's props.
+     *
+     * @param ref an {@code DailyPartitionCreator}.
+     * @param timeout timeout for the operation
+     * @throws ExecutionException if an exception was thrown during execution.
+     * @throws InterruptedException if the actor does not reply within the allotted timeout, or if the actor thread was
+     * interrupted for other reasons.
+     */
+    public static void start(
```

Comment:
Is there a reason that Actor must be explicitly started as opposed to just starting on the Akka onStart method?

Reply:
I originally had it started via `onStart` but did it this way so that I could synchronously start when the repository was opened, then any immediate errors when attempting to create the initial partition will bubble up to `Repository#open` instead of happening in the background.